client/me/security-2fa-sms-settings/index.jsx

```diff
@@ -89,44 +77,45 @@ const Security2faSMSSettings = createReactClass( {
 		}
 
 		return false;
-	},
+	}
 
-	onVerifyByApp: function ( event ) {
+	onVerifyByApp( event ) {
 		event.preventDefault();
 		this.verifyByApp = true;
 		this.submitSMSSettings();
-	},
+	}
 
-	onVerifyBySMS: function ( event ) {
+	onVerifyBySMS( event ) {
 		event.preventDefault();
 		this.verifyByApp = false;
 		this.submitSMSSettings();
-	},
-
-	/**
-	 * Note:  We purposely do not use form-base's submitForm so that we can
-	 * manage Notices ourselves
-	 */
-	submitSMSSettings: function () {
-		if ( ! this.refs.phoneInput ) {
-			return;
-		}
+	}
 
+	async submitSMSSettings() {
 		const phoneNumber = this.state.phoneNumber;
 
 		if ( ! phoneNumber.isValid ) {
 			this.setState( { lastError: phoneNumber.validation } );
 			return;
 		}
 
-		this.props.userSettings.updateSetting( 'two_step_sms_phone_number', phoneNumber.phoneNumber );
-		this.props.userSettings.updateSetting( 'two_step_sms_country', phoneNumber.countryCode );
+		if (
+			phoneNumber.countryCode === this.props.userSettings.two_step_sms_country &&
+			phoneNumber.phoneNumber === this.props.userSettings.two_step_sms_phone_number
+		) {
+			this.onSubmitResponse();
```

Comment:
To confirm, behind the scenes we do nothing, but for the user, we pretend that we're saving those, right?

Reply:
We don't pretend we're saving these settings as there's no _Save_ button or anything like that but buttons which introduce the next step (Verify by…). I guess it's more like _"Taking you to the next step and by the way we saved your settings in the meantime"_.